### PR TITLE
os/wqueue : Refactor work queue

### PR DESCRIPTION
**Pull request by YiHyunJin**
os/wqueue : Check underflow in remaining time

If 'elapsed' is greater than 'cur_work->delay', there is an underflow.
So It must check.

os/wqueue : Modify the 'delay' type

Since the value of 'delay' uses clock, I changed the data type to clock_t.

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [e019accf47d662d170d4e403f3f1cd301f4a1b95] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [e019accf47d662d170d4e403f3f1cd301f4a1b95] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [caa051dd06f8e9ea76bbf635b8d91e832d15a0a9] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [caa051dd06f8e9ea76bbf635b8d91e832d15a0a9] - Code Rule Check OK.

File: os/include/tinyara/wqueue.h [hunk 1]
```diff
@@ -387,7 +387,7 @@ int work_lpstart(void);
  *
  ****************************************************************************/
 
-int work_queue(int qid, FAR struct work_s *work, worker_t worker, FAR void *arg, uint32_t delay);
+int work_queue(int qid, FAR struct work_s *work, worker_t worker, FAR void *arg, clock_t delay);
```
Comment (SeonghoByeon) [inline os/include/tinyara/wqueue.h]: If you change the input type of existing API, it might occur compile warnings. Am I right?

Comment (YiHyunJin) [reply os/include/tinyara/wqueue.h]: I think there will be no compile  warning because clock_T is uint32_t or uint64_t

**Pull request merged**